integrations/gcp/gcp_core/search/resource_searches.py

```diff
@@ -25,169 +24,186 @@
     parse_protobuf_messages,
     parse_latest_resource_from_asset,
 )
-from gcp_core.search.utils import retry_policy
+from gcp_core.search.utils import paginated_query, DEFAULT_REQUEST_TIMEOUT
 
 
 async def search_all_resources(
-    project_data: dict[str, Any], asset_type: str
+    project_data: dict[str, Any], asset_type: str, **kwargs: Any
 ) -> ASYNC_GENERATOR_RESYNC_TYPE:
-    async for resources in search_all_resources_in_project(project_data, asset_type):
+    async for resources in search_all_resources_in_project(
+        project_data, asset_type, **kwargs
+    ):
         yield resources
 
 
 async def search_all_resources_in_project(
-    project: dict[str, Any], asset_type: str, asset_name: str | None = None
+    project: dict[str, Any],
+    asset_type: str,
+    asset_name: str | None = None,
+    **kwargs: Any,
 ) -> ASYNC_GENERATOR_RESYNC_TYPE:
     """
     List of supported assets: https://cloud.google.com/asset-inventory/docs/supported-asset-types
     Search for resources that the caller has ``cloudasset.assets.searchAllResources`` permission on within the project's scope.
     """
-    project_name = project["name"]
-    logger.info(f"Searching all {asset_type}'s in project {project_name}")
-    async with AssetServiceAsyncClient() as async_assets_client:
+
+    def parse_asset_response(response: Any) -> list[dict[Any, Any]]:
+        assets = typing.cast(list[AssetData], parse_protobuf_messages(response.results))
+        latest_resources = [
+            {
+                **parse_latest_resource_from_asset(asset),
+                EXTRA_PROJECT_FIELD: project,
+            }
+            for asset in assets
+        ]
+        return latest_resources
+
+    async with kwargs["asset_semaphore"]:
```

Comment:
Not sure we need the semaphore. We didn't encounter any problems with concurrency, the ratelimiter should suffice here